### PR TITLE
Updated the digital signatures section for FIPS 186-5

### PR DESCRIPTION
**Pull request by nmav**
This removes DSA, that is today a legacy algorithm not part of the FIPS 186-5 (DSS) standard. It adds EdDSA and mentions FIPS 186-5 as a google search for DSS brings historic documents not the current standard.

### CODE REVIEW

File: secure_software_development_fundamentals.md [hunk 1]
```diff
@@ -4569,9 +4569,9 @@ A widely-used public key algorithm is the RSA algorithm, which *can* be used for
...
+A whole family of algorithms are called *elliptic curve cryptography*; these are algorithms that are based on complex math involving elliptic curves. These algorithms require far shorter key lengths for equivalent cryptographic strength, and that is a significant advantage. Historically, elliptic curve cryptography involved a minefield of patents, but over the years many of those patents have expired and so elliptic curve cryptography has become more common. A widely-used and respected algorithm for key exchange is X25519 and EdDSA for Digital Signatures; a related protocol called ECIES combines X25519 key exchange with a symmetric key algorithm (for more details, see [*Seriously, stop using RSA*](https://blog.trailofbits.com/2019/07/08/fuck-rsa/), 2019).
```
Comment (david-a-wheeler) [inline secure_software_development_fundamentals.md:4572]: The text here is overly complicated; don't swap order in the middle. Specifically, change:

> A widely-used and respected algorithm for key exchange is X25519 and EdDSA for Digital Signatures;

to:

> A widely-used and respected elliptic curve algorithm for key exchange is X25519, while for digital signature it's EdDSA.

Also:

> a related protocol called ECIES combines Curve25519 key exchange with a symmetric key algorithm

I can't find evidence it *has* to always be Curve25519. I suggest making it general as follows:

> A related protocol called ECIES combines elliptic curve key exchange with a symmetric key algorithm

**Comment by david-a-wheeler** [comment]
Thanks for the feedback! Please see my comments.

File: secure_software_development_fundamentals.md (hunk 1 above)
Comment (nmav) [reply secure_software_development_fundamentals.md]: Updated.

**Comment by nmav** [comment]
Any updates regarding the inclusion of this PR? Removing DSA is essential for a modern cryptographic intro training since it’s no longer recommended by any active standards.

File: secure_software_development_fundamentals.md (hunk 1 above)
Comment (david-a-wheeler) [reply secure_software_development_fundamentals.md]: Almost. It's still missing "while for digital signature it's EdDSA." But I can fix that separately. Thanks so much for the recommended changes!

**Comment by david-a-wheeler** [comment]
Thank you so much, sorry for the delay.

**Pull request merged**